src/views/Timeline.vue

```diff
@@ -143,47 +123,69 @@ export default {
 	},
 
 	methods: {
-		async fetchContent() {
-			// only one simultaneous page load
-			if (this.loadingPage) {
-				return
+		async getContent(params) {
+			if (this.done) {
+				return Promise.resolve([])
 			}
 
 			// cancel any pending requests
 			if (this.cancelRequest) {
 				this.cancelRequest('Changed view')
 			}
 
-			// if we don't already have some cached data let's show a loader
-			if (this.timeline.length === 0) {
-				this.$emit('update:loading', true)
-			}
-			this.error = null
-			this.loadingPage = true
-
-			// init cancellable request
+			// done loading even with errors
 			const { request, cancel } = cancelableRequest(getPhotos)
 			this.cancelRequest = cancel
 
 			try {
-				// get content and current folder info
 				const files = await request(this.onlyFavorites, {
-					page: this.page,
-					perPage: this.gridConfig.count * 5, // we load 5 rows,
+					page: params.offset,
+					perPage: 30,
 				})
-				this.$store.dispatch('updateTimeline', files)
-				this.$store.dispatch('appendFiles', files)
 
-				// next time we load this script, we load the next page if the list returned
-				if (files.length === this.gridConfig.count * 5) {
-					this.page++
-				} else {
-					console.debug('We loaded the last page')
+				if (files.length === params.offset === 0) {
+					this.isEmpty = true
+				}
+
+				if (files.length !== 30) {
 					this.done = true
 				}
 
-				// return for the viewer loadMore method
-				return files
+				return files.flatMap((file, index) => {
+					const finalArray = []
+					const currentSection = this.getFormatedDate(file.lastmod, 'YYYY MMMM')
+					if (
+						currentSection
+						!== this.getFormatedDate(files[index - 1]?.lastmod, 'YYYY MMMM')
+						&& (this.lastSection !== currentSection)
+					) {
+						finalArray.push({
+							id: `title-${index}`,
+							injected: {
+								year: this.getFormatedDate(file.lastmod, 'YYYY'),
+								month: this.getFormatedDate(file.lastmod, 'MMMM'),
+							},
+							height: 90,
+							columnSpan: 0,
+							newRow: true,
+							renderComponent: SeparatorVirtualGrid,
+						})
+						this.lastSection = currentSection
+					}
+					finalArray.push({
+						id: `img-${file.fileid}`,
+						injected: {
+							...file,
+							list: this.$refs.virtualgrid.getCurrentItems,
+							loadMore: this.$refs.virtualgrid.loadMoreData,
+						},
+						width: 256,
+						height: 256,
+						columnSpan: 1,
+						renderComponent: FileVirtualGrid,
+					})
+					return finalArray
+				})
```

Comment:
I would love a little bit more comments here, so anyone can follow :see_no_evil: 
I prefer to have more comments than needed so at least it's clear :v:

Reply:
Sure ok

Reply:
I added comments